.github/workflows/ci.yml

```diff
@@ -0,0 +1,28 @@
+name: CI
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
```

Comment:
> if no activity types are specified, the workflow runs when a pull request is opened or reopened or when the head branch of the pull request is updated.

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request